compiler/z/codegen/S390Instruction.cpp

```diff
@@ -2392,21 +2392,7 @@ TR::S390RILInstruction::generateBinaryEncoding()
          {
          if (cg()->hasCodeCacheSwitched())
             {
-            TR::SymbolReference *calleeSymRef = NULL;
-
-            calleeSymRef = getSymbolReference();
-
-            if (calleeSymRef != NULL)
-               {
-               if (calleeSymRef->getReferenceNumber()>=TR_S390numRuntimeHelpers)
-                  cg()->fe()->reserveTrampolineIfNecessary(comp, calleeSymRef, true);
-               }
-            else
-               {
-#ifdef DEBUG
-               printf("Missing possible re-reservation for trampolines.\n");
-#endif
-               }
+            cg()->redoTrampolineReservationIfNecessary(this, getSymbolReference());
```

Comment:
On Z, the new common logic handles a couple more cases that the current implementation here doesn't (likely because those situations don't arise).  Namely, handling the case where the destination of the call is a label.  I believe the commoned logic should still be functionally correct.

Reply:
Thanks for consolidating. Yes this should be fine.